lib/src/screens/session/disclosure/bloc/disclosure_permission_bloc.dart

```diff
@@ -459,12 +459,24 @@ class DisclosurePermissionBloc extends Bloc<DisclosurePermissionBlocEvent, Discl
 
       // Check whether an issue wizard is needed to bootstrap the session.
       final choices = candidates.map(
-        (i, discon) => MapEntry(
-          i,
-          discon
-              .firstWhereOrNull((con) => con.every((cred) => cred is ChoosableDisclosureCredential))
-              ?.cast<ChoosableDisclosureCredential>(),
-        ),
+        (i, discon) {
+          final choosableCons = discon.where(
+            (con) => con.every((cred) => cred is ChoosableDisclosureCredential),
+          );
+
+          final Con<DisclosureCredential>? preferredCon =
+              // Always prefer an option that is not revoked / expired
+              choosableCons.firstWhereOrNull(
+                    (con) => con.every((cred) => !cred.revoked && !cred.expired),
```

Comment:
You can use `cred.valid` here if I'm right. This getter is a shortcut for this exact expression.